internal/recover/recover.go

```diff
@@ -144,43 +396,44 @@ func MaybeUnpackRecoveryTarball(filesystem *sys.OS) error {
 		return err
 	}
 
-	// sanity check: valid cluster.yaml in the incoming DB dir
-	clusterYamlPath := path.Join(unpackDir, "cluster.yaml")
-	incomingNodeInfo, err := dumpYamlNodeStore(clusterYamlPath)
-	if err != nil {
-		return err
-	}
-
-	// use the local info.yaml so that the dqlite ID is preserved on each
-	// cluster member
+	// We need to set the local info.yaml address with the (possibly changed)
+	// incoming address for this member.
 	localInfoYamlPath := path.Join(filesystem.DatabaseDir, "info.yaml")
 	recoveryInfoYamlPath := path.Join(unpackDir, "info.yaml")
 
-	localInfoYaml, err := os.ReadFile(localInfoYamlPath)
+	var localInfo dqlite.NodeInfo
+	err = readYaml(localInfoYamlPath, &localInfo)
 	if err != nil {
 		return err
 	}
 
-	var localInfo dqlite.NodeInfo
-	err = yaml.Unmarshal(localInfoYaml, &localInfo)
+	var incomingMembers []cluster.DqliteMember
+	err = readYaml(recoveryYamlPath, &incomingMembers)
 	if err != nil {
-		return fmt.Errorf("invalid %q", localInfoYamlPath)
+		return nil
 	}
 
 	found := false
-	for _, incomingInfo := range incomingNodeInfo {
-		found = localInfo.ID == incomingInfo.ID
+	for _, incomingInfo := range incomingMembers {
+		found = localInfo.ID == incomingInfo.DqliteID
 
 		if found {
+			localInfo.Address = incomingInfo.Address
 			break
 		}
 	}
 
 	if !found {
-		return fmt.Errorf("missing local cluster member in incoming cluster.yaml")
+		return fmt.Errorf("missing local cluster member in incoming recovery.yaml")
```

Comment:
Errors should start with an uppercase letter

Reply:
We should implement a lint check for this in lxd/microcloud/microcluster (if not already done)